modules/servicetemplate/src/main/java/org/eclipse/xpanse/modules/servicetemplate/ServiceTemplateManage.java

```diff
@@ -103,11 +103,14 @@ public ServiceTemplateEntity updateServiceTemplate(UUID id, Ocl ocl) {
                 && Objects.nonNull(ocl.getServiceConfigurationManage())) {
             serviceConfigurationParameterValidator.validateServiceConfigurationParameters(ocl);
         }
-        validateServiceDeployment(ocl.getDeployment(), existingTemplate);
-        existingTemplate.setOcl(ocl);
-        setServiceRegistrationState(existingTemplate);
+        existingTemplate.setIsUpdatePending(true);
         ServiceTemplateEntity updatedServiceTemplate =
                 templateStorage.storeAndFlush(existingTemplate);
+        validateServiceDeployment(ocl.getDeployment(), updatedServiceTemplate);
+        updatedServiceTemplate.setOcl(ocl);
+        setServiceTemplateRegistrationState(updatedServiceTemplate);
+        updatedServiceTemplate.setIsUpdatePending(false);
```

Comment:
why do we immediately change this value to 'false'?

Reply:
This will be updated after the table  service_template_history created.